File: v1/java/accumulate/src/main/java/Accumulate.java
import java.util.LinkedList;
import java.util.List;
import java.util.function.Function;

public class Accumulate {
    public static <T> List<T> accumulate(List<T> input, Function<T, T> map) {
        List<T> results = new LinkedList<T>();

        for (T element: input) {
            results.add(map.apply(element));
        }

        return results;
    }
}

File: kotlin/accumulate/src/main/kotlin/Accumulate.kt
import java.util.*

object Accumulate {
    fun <T, V> accumulate(input: List<T>, callback: (T) -> V): List<V> {
        val results = LinkedList<V>()

        for (x in input) {
            results.add(callback(x))
        }

        return results
    }
}